openapi-core/src/main/java/io/ballerina/openapi/core/generators/service/ParametersGenerator.java

```diff
@@ -137,6 +142,19 @@ public void generateResourcesInputs(Map.Entry<PathItem.HttpMethod, Operation> op
                         }
                     }
                 }
+
+                if (param != null) {
+                    String parameterName = param instanceof RequiredParameterNode ?
+                            ((RequiredParameterNode) param).paramName().get().text() :
+                            ((DefaultableParameterNode) param).paramName().get().text();
```

Comment:
Casting to `DefaultableParameterNode` in else case might not be safe in here if there will be be rest parameters. Shall we make sure that rest parameters are not allowed for resource functions?

Reply:
According to the implementation here in openapi-to-ballerina-service generation, any parameter given in the openapi has mapped to either required parameter of defaultable  parameter. AFAIK in the current implementation there is no openapi parameters schema represented as rest type parameter in Ballerina. 

@lnash94 appreciate you thoughts too.

Reply:
Only required and default parameters are mapped in openapi-to-ballerina service/client generations. Although Resource functions can have rest parameters as the path parameters in ballerina but still need to investigate a representation for the rest parameters in OpenAPI specification under the parameter schemas.